pkg/cmd/cli/secrets/basicauth.go

```diff
@@ -162,6 +162,20 @@ func (o *CreateBasicAuthSecretOptions) Complete(f *kcmdutil.Factory, args []stri
 	}
 	o.SecretName = args[0]
 
+	if o.PromptForPassword {
+		if len(o.Password) != 0 {
+			return errors.New("must provide either --prompt or --password flag")
+		}
+		if cmdutil.IsTerminal(o.Reader) {
```

Comment:
nit: Make this `!cmdutil.IsTerminal(o.Reader)` and drop the else clause: one golint error less:)